detoks/src/main/java/nl/tudelft/trustchain/detoks/LightweightTxApp.kt

```diff
@@ -0,0 +1,114 @@
+package nl.tudelft.trustchain.detoks
+
+import com.squareup.sqldelight.sqlite.driver.JdbcSqliteDriver
+import com.squareup.sqldelight.db.SqlDriver
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import nl.tudelft.ipv8.*
+import nl.tudelft.ipv8.attestation.trustchain.ProposalBlockBuilder
+import nl.tudelft.ipv8.attestation.trustchain.store.TrustChainSQLiteStore
+import nl.tudelft.ipv8.keyvault.JavaCryptoProvider
+import nl.tudelft.ipv8.messaging.EndpointAggregator
+import nl.tudelft.ipv8.messaging.udp.UdpEndpoint
+import nl.tudelft.ipv8.peerdiscovery.strategy.RandomWalk
+import nl.tudelft.ipv8.sqldelight.Database
+import java.net.InetAddress
+import java.util.*
+import kotlin.math.roundToInt
+
+class DemoTransactionApp {
+    private val scope = CoroutineScope(Dispatchers.Default)
+
+    fun run() {
+        startIpv8()
+    }
+
+    private fun createCommunity(): OverlayConfiguration<TransactionEngine> {
+        val randomWalk = RandomWalk.Factory(timeout = 3.0, peers = 20)
+        return OverlayConfiguration(
+            TransactionEngine.Factory("c86a7db45eb3563ae047639817baec4db2bc7c25"),
+            listOf(randomWalk)
+        )
+    }
+
+    private fun startIpv8() {
+        val myKey = JavaCryptoProvider.generateKey()
+        val myPeer = Peer(myKey)
+        val udpEndpoint = UdpEndpoint(8080, InetAddress.getByName("0.0.0.0"))
+        val endpoint = EndpointAggregator(udpEndpoint, null)
+
+        val config = IPv8Configuration(overlays = listOf(
+            createCommunity()
+        ), walkerInterval = 1.0)
+
+        // storage
+        val driver: SqlDriver = JdbcSqliteDriver(JdbcSqliteDriver.IN_MEMORY)
+        Database.Schema.create(driver)
+        val store = TrustChainSQLiteStore(Database(driver))
+
+        val ipv8 = IPv8(endpoint, config, myPeer)
+        ipv8.start()
+
+        val numberOfTransactions = 1
+        scope.launch {
+            while (true) {
+                val overlay = ipv8.overlays.values.toList()[0]
+                printPeersInfo(overlay)
+                println("===")
```

Comment:
remove

Reply:
Thanks for the comment. Could you be a bit more specific here? Are you referring to the entire `while (true)` statement? Or to the `printPeersInfo(overlay)` part?